cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -481,18 +481,41 @@ public TableInfo getTableInfo(@Nullable String database, String table)
 
     try {
       String db = database == null ? "default" : database;
+
       Table tableInfo = getMetaStoreClient().getTable(db, table);
+      List<FieldSchema> tableFields = tableInfo.getSd().getCols();
+      // for whatever reason, it seems like the table columns for partitioned tables are not present
+      // in the storage descriptor. If columns are missing, do a separate call for schema.
+      if (tableFields == null || tableFields.isEmpty()) {
+        // don't call .getSchema()... class not found exception if we do in the thrift code...
+        tableFields = getMetaStoreClient().getFields(db, table);
+      }
+
       ImmutableList.Builder<TableInfo.ColumnInfo> schemaBuilder = ImmutableList.builder();
-      ImmutableList.Builder<TableInfo.ColumnInfo> partitionKeysBuilder = ImmutableList.builder();
-      for (FieldSchema column : tableInfo.getSd().getCols()) {
+      for (FieldSchema column : tableFields) {
         schemaBuilder.add(new TableInfo.ColumnInfo(column.getName(), column.getType(), column.getComment()));
       }
+
+      ImmutableList.Builder<TableInfo.ColumnInfo> partitionKeysBuilder = ImmutableList.builder();
       for (FieldSchema column : tableInfo.getPartitionKeys()) {
-        partitionKeysBuilder.add(new TableInfo.ColumnInfo(column.getName(), column.getType(),
-                                                                     column.getComment()));
+        TableInfo.ColumnInfo columnInfo = new TableInfo.ColumnInfo(column.getName(), column.getType(),
+                                                                   column.getComment());
+        partitionKeysBuilder.add(columnInfo);
+        // add partition keys to the schema as well, since they show up when you do a 'describe <table>' command.
+        schemaBuilder.add(columnInfo);
+      }
```

Comment:
I wonder what happens if a column of the schema is used as a partitioning key. I assume you will see that column twice here? Perhaps you should filter such duplicates?